libcodechecker/libhandlers/server.py

```diff
@@ -342,6 +369,193 @@ def arg_match(options):
+    """
+
+    LOG.debug("Checking config database")
+    migration_root = context.run_migration_root
+
+    engine = cfg_sql_server.create_engine()
+    config_session = sessionmaker(bind=engine)
+    sess = config_session()
+
+    products = sess.query(ORMProduct).all()
+
+    prod_status = {}
+    for pd in products:
+        db = database.SQLServer.from_connection_string(pd.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        ret = db.connect()
+        prod_status[pd.endpoint] = ret
+
+    sess.commit()
+    sess.close()
+    engine.dispose()
+
+    return prod_status
+
+
+def __db_status_check(cfg_sql_server, context, product_name=None):
+    """
+    Check and print database statuses for the given product.
+    """
+    if not product_name:
+        return 0
+
+    print(product_name)
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
+        if not avail:
+            LOG.error("No product was found with this endpoint: " +
+                      str(product_name))
+            return 1
+
+        prod_statuses = {k: v for k, v in prod_statuses.items()
+                         if k == product_name}
+
+    print_prod_status(prod_statuses)
+    return 0
+
+
+def __db_migration(cfg_sql_server, context, product_to_upgrade):
+    """
+    Handle database management.
+    Schema checking and migration.
+    """
+    LOG.info("Migrating schema for " + str(product_to_upgrade))
+    product_name = product_to_upgrade
+
+    prod_statuses = check_product_db_status(cfg_sql_server, context)
+    prod_to_upgrade = []
+
+    if product_name != 'all':
+        avail = prod_statuses.get(product_name)
+        if not avail:
+            LOG.error("No product was found with this endpoint: " +
+                      product_name)
+            return 1
+        prod_to_upgrade.append(product_name)
+    else:
+        prod_to_upgrade = list(prod_statuses.keys())
+
+    migration_root = context.run_migration_root
+
+    for prod in prod_to_upgrade:
+        LOG.info("========================")
+        LOG.info("Upgrading: " + prod)
+        engine = cfg_sql_server.create_engine()
+        config_session = sessionmaker(bind=engine)
+        sess = config_session()
+
+        product = sess.query(ORMProduct).filter(
+                ORMProduct.endpoint == prod).first()
+        db = database.SQLServer.from_connection_string(product.connection,
+                                                       RUN_META,
+                                                       migration_root,
+                                                       interactive=False)
+
+        db_status = db.connect()
+
+        if db_status == DBStatus.OK:
```

Comment:
To print the database status use the helper function what I have mentioned above at `handle_connect_ret`. Handle special database status cases after the log. 
```py
Log.info(util.db_status_to_str(db_status))

if db_status == DBStatus.SCHEMA_MISSING:
    question = ...
elif db_status == DBStatus.SCHEMA_MISMATCH_OK_MIGRATION:
    question = ...
```